packages/usa-input-mask/src/index.js

```diff
@@ -0,0 +1,162 @@
+      const l = inputs.length;
+
+      for (i = 0; i < l; i += 1) {
+        masking.createShell(inputs[i]);
+      }
+    },
+
+    // replaces each masked input with a shell containing the input and it's mask.
+    createShell(input) {
+      let text = "";
+      const placeholder = input.getAttribute(`${PLACEHOLDER}`);
+
+      if (
+        placeholder !== null &&
+        placeholder.length > 0 &&
+        placeholder !== " "
+      ) {
+        input.setAttribute("maxlength", placeholder.length);
+        input.setAttribute("data-placeholder", placeholder);
+        input.removeAttribute(`${PLACEHOLDER}`);
+      } else {
+        return;
+      }
+
+      // This section still needs some rework
+      text = `<span class="${SHELL}"><span aria-hidden="true" id="${input.id}Mask"><i></i>${placeholder}</span>${input.outerHTML}</span>`;
+
+      input.outerHTML = text; // eslint-disable-line no-param-reassign, no-unsanitized/property
+    },
+
+    setValueOfMask(e) {
+      const { value } = e.target;
+      const placeholder = e.target.getAttribute("data-placeholder");
+      const placeholderVal = `${placeholder.substr(value.length)}`;
+
+      const theIEl = document.createElement("i");
+      theIEl.textContent = value;
+      return [theIEl, placeholderVal];
+    },
+
+    // add event listeners
+    activateMasking(inputs) {
+      let i;
+      let l;
+
+      for (i = 0, l = inputs.length; i < l; i += 1) {
+        masking.maskedInputs[i].addEventListener(
+          "keyup",
+          (e) => {
+            masking.handleValueChange(e);
+          },
+          false
+        );
+      }
+    },
+
+    handleValueChange(e) {
+      const id = e.target.getAttribute("id");
+
+      switch (
+        e.keyCode // allows navigating thru input
+      ) {
+        case 20: // caplocks
+        case 17: // control
+        case 18: // option
+        case 16: // shift
+        case 37: // arrow keys
+        case 38:
+        case 39:
+        case 40:
+        case 9: // tab (let blur handle tab)
+          return;
+        // no default
+      }
+
+      document.getElementById(id).value = masking.handleCurrentValue(e);
+      const maskVal = masking.setValueOfMask(e);
+      const maskEl = document.getElementById(`${id}Mask`);
+      maskEl.innerHTML = "";
+      maskEl.replaceChildren(maskVal[0], maskVal[1]);
+    },
+
+    handleCurrentValue(e) {
+      const isCharsetPresent = e.target.getAttribute("data-charset");
+      const placeholder =
+        isCharsetPresent || e.target.getAttribute("data-placeholder");
+      const { value } = e.target;
+      const l = placeholder.length;
+      let newValue = "";
+      let i;
+      let j;
+      let isInt;
+      let isLetter;
+
+      // strip special characters
+      const strippedValue = isCharsetPresent
+        ? value.replace(/\W/g, "")
+        : value.replace(/\D/g, "");
+
+      for (i = 0, j = 0; i < l; i += 1) {
+        // eslint-disable-next-line no-restricted-globals
```

Comment:
What do `i` and `j` represent?

Reply:
i represents the location in the mask of input + characters and spaces
j represents how many characters have been typed in a given input

There might be a better or cleaner way to go about this, could refactor at some point